packages/medusa/src/loaders/helpers/routing/index.ts

```diff
@@ -0,0 +1,566 @@
+    config,
+  }: {
+    config?: MiddlewaresConfig
+  }): void {
+    if (!config?.routes) {
+      log({
+        activityId: this.activityId,
+        message: `No middleware routes found. Skipping middleware application.`,
+      })
+
+      return
+    }
+
+    for (const route of config.routes) {
+      if (!route.matcher) {
+        throw new Error(
+          `Route is missing a \`matcher\` field. The 'matcher' field is required when applying middleware to this route.`
+        )
+      }
+    }
+  }
+
+  /**
+   * Take care of replacing the special path segments
+   * to an express specific path segment
+   *
+   * @param route - The route to parse
+   *
+   * @example
+   * "/admin/orders/[id]/index.ts" => "/admin/orders/:id/index.ts"
+   */
+  protected parseRoute(route: string): string {
+    let route_ = route
+
+    for (const config of Object.entries(pathSegmentReplacer)) {
+      const [searchFor, replacedByFn] = config
+      const replacer = new RegExp(searchFor, "g")
+
+      const matches = [...route_.matchAll(replacer)]
+
+      const parameters = new Set()
+
+      for (const match of matches) {
+        if (match?.[1] && !Number.isInteger(match?.[1])) {
+          if (parameters.has(match?.[1])) {
+            log({
+              activityId: this.activityId,
+              message: `Duplicate parameters found in route ${route} (${match?.[1]})`,
+            })
+
+            throw new Error(
+              `Duplicate parameters found in route ${route} (${match?.[1]}). Make sure that all parameters are unique.`
+            )
+          }
+
+          parameters.add(match?.[1])
+        }
+
+        route_ = route_.replace(match[0], replacedByFn(match?.[1]))
+      }
+
+      const extension = extname(route_)
+      if (extension) {
+        route_ = route_.replace(extension, "")
+      }
+    }
+
+    route = route_
+
+    return route
+  }
+
+  /**
+   * Load the file content from a descriptor and retrieve the verbs and handlers
+   * to be assigned to the descriptor
+   *
+   * @return {Promise<void>}
+   */
+  protected async createRoutesConfig(): Promise<void> {
+    await Promise.all(
+      [...this.routesMap.values()].map(async (descriptor: RouteDescriptor) => {
+        const absolutePath = descriptor.absolutePath
+
+        return await import(absolutePath).then((imp) => {
+          const map = this.routesMap
+
+          const config: {
+            routes: RouteConfig[]
+            shouldRequireAuth?: boolean
+          } = {
+            routes: [],
+            shouldRequireAuth: false,
+          }
+
+          /**
+           * If the developer has not exported the authenticate flag
+           * we default to true.
+           */
+          const shouldRequireAuth =
+            imp[AUTHTHENTICATE] !== undefined
+              ? (imp[AUTHTHENTICATE] as boolean)
+              : true
+
+          if (
+            shouldRequireAuth &&
+            absolutePath.includes(join("api", "admin"))
```

Comment:
question: for stores route we don't support auth? I think it would be nice to be able to set the flag on the store route file and get the customerAuth applied automatically as well no? for store route it would be false by default. wdyt?